### PR TITLE
Fix 02720_row_policy_column_with_dots

### PR DESCRIPTION
**Pull request by canhld94**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Test data should be deterministic.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 009c0dc136b8e74c41784011d77e85ce18ff7e8d with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59453/009c0dc136b8e74c41784011d77e85ce18ff7e8d/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59453/009c0dc136b8e74c41784011d77e85ce18ff7e8d/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by jkartseva** [comment]
Thanks for fixing it!
Instead of magic constants, can we use something like:

``` sql
INSERT INTO table_with_dot_column select date_add(YEAR, 1, NOW()), 'Hello', 'World';
```

``` sql
CREATE ROW POLICY IF NOT EXISTS row_policy ON table_with_dot_column USING toDate(date) >= today() - 30 TO ALL;
SELECT count(*) FROM table_with_dot_column WHERE toDate(date) > today();
SELECT count(*) FROM table_with_dot_column; 
```

**Comment by azat** [comment]
FWIW this breaks fast test (in other words - CI), so it will be great if this will be merged as soon as possible

File: tests/queries/0_stateless/02720_row_policy_column_with_dots.sql [hunk 1]
```diff
@@ -1,6 +1,6 @@
-CREATE table if not exists table_with_dot_column (date Date, regular_column String, `other_column.2` String) ENGINE = MergeTree() ORDER BY date;
-INSERT INTO table_with_dot_column select '2020-01-01', 'Hello', 'World';
-INSERT INTO table_with_dot_column select '2024-01-01', 'Hello', 'World';
+CREATE TABLE IF NOT EXISTS table_with_dot_column (date Date, regular_column String, `other_column.2` String) ENGINE = MergeTree() ORDER BY date;
+INSERT INTO table_with_dot_column SELECT '2020-01-01', 'Hello', 'World';
+INSERT INTO table_with_dot_column SELECT toDate(now() + 48*3600), 'Hello', 'World';
```
Comment (antonio2368) [inline tests/queries/0_stateless/02720_row_policy_column_with_dots.sql:3]: just FYI this will also work
```sql
SELECT toDate(now() + INTERVAL 2 DAYS)
```

**Pull request merged**